app/core/models.py

```diff
@@ -329,3 +329,23 @@ class Sdg(AbstractBaseModel):
 
     def __str__(self):
         return f"{self.name}"
+
+
+class ProjectSponsorPartnerXref(AbstractBaseModel):
+    """
+    Sponsor/partner relationships stored in this table are project-dependent.
+    They cannot be both a sponsor and a partner for the same project,
+    so if is_sponsor is FALSE they are a project partner,
+    if is_sponsor is TRUE they are a project sponsor.
+    """
+
+    partner_id = models.ForeignKey(SponsorPartner, on_delete=models.CASCADE)
+    project_id = models.ForeignKey(Project, on_delete=models.CASCADE)
+    ended_at = models.DateTimeField("Ended at", null=True, blank=True)
+    is_sponsor = models.BooleanField(null=True)
```

Comment:
Yep totally agreed, tell me when the issues are ready!!

Reply:
@freaky4wrld Can you just make the field required for now.

There's another thing I didn't consider since I'm not really a database person. Do we need the partner-project association to be unique? I'm thinking yes and that we need to add a uniqueness constraint to the model.

Reply:
Thanks! I think we do need to add a uniqueness constraint. Here's the generated suggestion as I was trying to add it myself. It seems to work when I tried to add the same relationship twice. I looked up the `unique_together` constraint that I remember hearing about, but the django docs now recommends using this more generic syntax.
```python
class ProjectSponsorPartnerXref(AbstractBaseModel):
    ...

    class Meta:
        constraints = [
            models.UniqueConstraint(
                fields=["project", "partner"], name="unique_project_partner"
            )
        ]
```